### PR TITLE
[C-1626] Connect phantom wallet-connect with common sagas

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Moves common wallet-sagas to web common
Uses common saga helpers in phantom flow

TODO: modularize and hook up wallet-connect eth sagas

### Dragons

The transaction appears to go through, and wallet is added on optimistic side, however the wallet connection doesn't seem to add on user..

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-1626-phantom-wallet-connect

File: packages/mobile/src/components/navigation-container/NavigationContainer.tsx [hunk 1]
```diff
@@ -54,80 +54,84 @@ const NavigationContainer = (props: NavigationContainerProps) => {
...
-                        Playlists: 'playlists',
-                        Reposts: 'reposts',
-                        Collectibles: 'collectibles'
-                      }
-                    },
-                    SettingsScreen: {
-                      path: 'settings'
-                    }
+                    Wallets: 'wallets'
                   }
```
Comment (raymondjacobson) [inline packages/mobile/src/components/navigation-container/NavigationContainer.tsx:135]: was this primarily just a reoder?

File: packages/mobile/src/store/wallet-connect/sagas/connectNewWalletSaga.ts [hunk 2]
```diff
@@ -47,6 +49,14 @@ function* connectNewWalletAsync(action: ConnectNewWalletAction) {
       const connectData = decryptPayload(data, nonce, sharedSecretDapp)
       const { session, public_key } = connectData
 
+      const isNewWallet = yield* checkIsNewWallet(public_key, Chain.Sol)
+      if (!isNewWallet) return
+
+      const { balance, collectibleCount } = yield* getWalletInfo(
+        public_key,
+        Chain.Sol
```
Comment (raymondjacobson) [inline packages/mobile/src/store/wallet-connect/sagas/connectNewWalletSaga.ts:57]: This will need to be cased right?

**Review by raymondjacobson — APPROVED** [review]
so nice!

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-1626-phantom-wallet-connect

File: packages/mobile/src/store/wallet-connect/sagas/connectNewWalletSaga.ts (hunk 2 above)
Comment (dylanjeffers) [reply packages/mobile/src/store/wallet-connect/sagas/connectNewWalletSaga.ts:56]: would it for phantom?

Comment (raymondjacobson) [reply packages/mobile/src/store/wallet-connect/sagas/connectNewWalletSaga.ts:56]: nope sorry, i didn't expand above lines. this is in the switch

**Pull request merged**